vscode-lean4/src/infoview.ts

```diff
@@ -228,17 +232,8 @@ export class InfoProvider implements Disposable {
             webviewPanel.webview.html = this.initialHtml();
 
             // this.client.initializeResult can be null if the server failed to start.
-            if (this.client.initializeResult) {
-                await webviewPanel.api.initialize(this.client.initializeResult, this.getLocation(editor))
```

Comment:
Much nicer, separating out the this.client.initializeResult until the separate serverRestarted event makes a lot of sense.